plugins/tiff/src/tiffpresentation.cc

```diff
@@ -389,6 +390,85 @@ void TiffPresentation::done()
   tif = NULL;
 }
 
+/**
+ * Method to override operator+= method
+ * Returns y if x is empty.
+ * Otherwise adds the values of the same key.
+ */
+PipetteLayerOperations::PipetteColor operator+=(PipetteLayerOperations::PipetteColor& x, const PipetteLayerOperations::PipetteColor& y)
+{
+  if(x.empty())
+  {
+    x = y;
+    return x;
+  }
+  for(auto const& elem : y)
+  {
+    x[elem.first] += elem.second;
+  }
+  return x;
+}
+
+/**
+ * Override operator/ method
+ * Divides each element inside the map by a constant
+ */
+PipetteLayerOperations::PipetteColor operator/(PipetteLayerOperations::PipetteColor x, const int y)
+{
+  for(auto elem : x)
+  {
+    x[elem.first] = elem.second / y;
+  }
+  return x;
+}
+
+////////////////////////////////////////////////////////////////////////
+// PipetteViewInterface
+////////////////////////////////////////////////////////////////////////
+PipetteLayerOperations::PipetteColor TiffPresentation::getAverages(Scroom::Utils::Rectangle<int> area)
+{
+  PipetteLayerOperations::Ptr pipetteLayerOperation = boost::dynamic_pointer_cast<PipetteLayerOperations>(this->ls[0]);
+  if(!pipetteLayerOperation)
+  {
+    return {};
+  }
+
+  Layer::Ptr bottomLayer = this->tbi->getBottomLayer();
+  PipetteLayerOperations::PipetteColor pipetteColors;
+
+  int totalPixels = area.getWidth() * area.getHeight();
+  
+  if(totalPixels == 0){
```

Comment:
In case of an empty selection, `PixelColors` will be an empty map, and hence you will not enter the body of the for-loop that does the division.

Reply:
I tested this again. If `totalPixels == 0` `pipetteColors` will not be an empty map, but a map of 0's. Then a division by 0 occurs which returns `-nans`.

Reply:
You are saying that if totalPixels is 0, we're still going to decompress a tile, and call `sumPixelValues()` on it. Why? If we don't need any pixels of that particular tile?

Reply:
Perhaps there is some miscommunication going on from my side. I meant that without the check and with `totalPixels == 0` the tile will get loaded. The check will make sure that if `totalPixels == 0` no tile will get loaded (as it will not reach the `for` loops).
 I am not sure if you are suggesting to take out the `if` check or if it is fine using this check. The check is needed because the `tile_pos_x_start` and `tile_pos_x_end` (and similarly for `y`)  cannot really distinguish whether a rectangle actually has an `area > 0` or not. They are just there to check to make sure to loop over the correct tiles.

Reply:
Hmm... Perhaps it is easiest for now to leave the check in.

Yes, I was indeed suggesting that if the code that computes which tiles are needed, returns more tiles than are actually needed, then it contains a bug.

That's obviously the case when we have zero pixels, and yet the code returns the one tile :wink: